nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/Conv1D.java

```diff
@@ -59,6 +59,7 @@
 public class Conv1D extends DynamicCustomOp {
 
     protected Conv1DConfig config;
+    private static final String INVALID_CONFIGURATION = "Invalid Conv1D configuration : s = %d p = %d ";
```

Comment:
Minor issue: Should always be ```%s``` - Preconditions doesn't need format specified for different types.
Everything eventually ends up using Object.toString() anyway for these.
Only valid format specifiers other than ```%s``` are these: https://github.com/deeplearning4j/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/util/NDArrayPreconditionsFormat.java#L24

Reply:
There are primitive data types in context of this message, no Objects. Should I wrap long to Long?

Reply:
No, no need to box integers to Number classes.
That happens automatically - but only if an exception is to be thrown.
https://github.com/deeplearning4j/deeplearning4j/blob/master/nd4j/nd4j-common/src/main/java/org/nd4j/base/Preconditions.java#L73
```
public static void throwEx(String message, Object... args) {
```

The idea with all of the (seeminly redundant) primitive overloads in Preconditions class is to avoid creating objects unless an exception is thrown as much as possible, to avoid unnecessary garbage generation.